### ISSUE TITLE
WP-CLI creates folder with .tmp extension

### ISSUE THREAD

**Issue by mAAdhaTTah** [wp-cli/wp-cli#1619]
This issue was reported here: https://github.com/mAAdhaTTah/WP-Gistpen/issues/24

```
wp plugin install https://github.com/mAAdhaTTah/WP-Gistpen/releases/download/0.5.0b1/wp-gistpen.zip
```

created a folder named `wp-gistpen.tmp` in my plugins dir. I have reproduced this one myself and can confirm I'm seeing this behavior.

**Comment by danielbachhuber** [comment]
I've reproduced as well. This actually appears to be a problem with `download_url()` in core. The file gets downloaded to `/tmp/wp-gistpen.tmp`, and `.tmp` becomes its new extension.

See https://core.trac.wordpress.org/ticket/30945

**Comment by mAAdhaTTah** [comment]
Awesome, thanks for your help.

**Comment by danielbachhuber** [comment]
This will be fixed in WP 4.2

**Comment by mAAdhaTTah** [comment]
Sweet, thanks.
